ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -974,6 +975,94 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
     return documents
 
 
+@with_traced_module
+def traced_base_tool_invoke(langchain, pin, func, instance, args, kwargs):
+    integration = langchain._datadog_integration
+    tool_input = get_argument_value(args, kwargs, 0, "input")
+    config = get_argument_value(args, kwargs, 1, "config", optional=True)
+
+    span = integration.trace(
+        pin,
+        "%s.%s.%s.%s" % (func.__module__, func.__class__.__name__, func.__name__, func.__self__.name),
+        interface_type="tool",
+    )
+
+    tool_output = None
+    try:
+        tool_attributes = ["name", "description", "metadata", "tags"]
+
+        for attribute in tool_attributes:
+            value = getattr(instance, attribute, None)
+            if isinstance(value, dict):
+                for key, meta_value in value.items():
+                    span.set_tag_str("langchain.request.tool.%s.%s" % (attribute, key), str(value))
+            elif isinstance(value, list):
+                for idx, tag in enumerate(value):
+                    span.set_tag_str("langchain.request.tool.%s.%d" % (attribute, idx), str(value))
+            elif value is not None:
+                span.set_tag_str("langchain.request.tool.%s" % attribute, str(value))
+
+        if integration.is_pc_sampled_span(span):
+            if tool_input:
+                span.set_tag_str("langchain.request.input", integration.trunc(str(tool_input)))
+            if config:
+                span.set_tag_str("langchain.request.config", json.dumps(config))
```

Comment:
This can always be tagged since it is not a direct I/O, doesn't need to be in the `is_pc_sampled_span()` block.

Reply:
Isn't a free field like this prone to grow large as well similar to the input ?

Reply:
Technically it's free but since it's not direct I/O I think we can be fine with that.